### DISCUSSION TITLE
How to capitalize the first letter?

### DISCUSSION THREAD

**Discussion by fovj** [L3MON4D3/LuaSnip#538] [Q&A] [answered]
`upper` and `lower` is fine. 

But I don't know if there is a corresponding way to capitalize the first letter.

```lua
  s(
    "us",
    fmt("const [{}, set{setter}] = useState({})", {
      i(1, "value"),
      i(2, "initialValue"),
      setter = l(l._1:upper(), 1)
    })
  ),
```

**Comment by L3MON4D3** [comment, marked as answer]
Just a bit of playing around with string-functions:
`setter = l(l._1:sub(1,1):upper() .. l._1:sub(2,-1)))`
(so extract first char, capitalize, append remaining chars)

**Reply by marcoSven** [reply]
I am using my vscode snippets. Is there an existing solution to get TextMate variable transformation working?

```
"useState": {
  "prefix": "useState",
  "body": [
    "const [${2:state}, set${2/(.*)/${1:/capitalize}/}] = useState$1(${3:initialState})"
  ]
},
```

**Reply by marcoSven** [reply]
I figured it out or at least I resolved it. The issue was related to "jsregexp" I think.

`build = "make install_jsregexp",` or running it manually, didn't result in any errors. 
I decided to remove and reinstall all my related plugins and now it works 🤷

**Reply by json-bateman** [reply]
In case someone stumbles upon this thread again, in the accepted answer it's missing the second argument in the lambda function, the code that works for me: 

```   
ls.s(
      "useS",
      fmt("const [{}, set{setter}] = useState({})", {
         i(1, "state"),
         i(2, "initialValue"),
         setter = l(l._1:sub(1, 1):upper() .. l._1:sub(2, -1), 1)
      })
   ),
```
